src/core/node.ts

```diff
@@ -31,15 +31,19 @@ export class Node {
         parent: Node | null,
         subpath: string,
         environment: any,
-        storedValue: any
+        initialValue: any,
+        createNewInstance: (initialValue: any) => any = identity,
+        finalizeNewInstance: (node: Node, initialValue: any) => void = noop
     ) {
         this.type = type
         this._parent = parent
         this.subpath = subpath
-        this.storedValue = storedValue
+        this.storedValue = createNewInstance(initialValue)
```

Comment:
@mweststrate Just seen :)
Btw it does'nt matter the order because afaik and if you watch the old code, createNewInstance is called before node creation, so has not any concern about environment theorically.

Swapped anyway to make code more readable!